.github/scripts/check-for-update.sh

```diff
@@ -1,178 +0,0 @@
-echo "setting up ssh repo"
-
-mkdir -p ~/.ssh
-echo "$SSH_KEY" > ~/.ssh/id_rsa
-chmod 600 ~/.ssh/id_rsa
-ssh-keyscan github.com >> ~/.ssh/known_hosts
-
-git config --global user.email "prismabots@gmail.com"
-git config --global user.name "Prismo"
-
-git remote add github "git@github.com:$GITHUB_REPOSITORY.git"
-git fetch github "$branch"
-git reset --hard "github/$branch"
-git checkout "github/$branch"
-
-# prepare script: read package.json but ignore workspace package.json files, redwood "web" and "api" package.json file
-pkg="var pkg=require('./package.json'); if (pkg.workspaces || pkg.name == '.prisma/client' || pkg.name == 'web' || pkg.name == 'api') { process.exit(0); }"
-
-# since GH actions are limited to 5 minute cron jobs, just run this continuously for 5 minutes
-minutes=5   # cron job runs each x minutes
-interval=10 # run each x seconds
-i=0
-count=$(((minutes * 60) / interval))
-echo ""
-echo "=========================="
-echo "running loop $count times"
-while [ $i -le $count ]; do
-  # increment to prevent forgetting incrementing, and also prevent overlapping with the next 5-minute job
-  i=$((i + 1))
-  echo ""
-  echo "=========================="
-  echo "run $i"
-
-  start=$(date "+%s")
-
-  dir=$(pwd)
-
-  echo "=========================="
-  echo "updating git checkout"
-  git fetch github "$branch"
-  git reset --hard "github/$branch"
-
-  echo "=========================="
-  echo "getting package version:"
-  v=$(bash .github/scripts/prisma-version.sh "$branch")
-  if [ -z "$v" ]
-  then
-        echo "Prisma version is empty: $v"
-        exit 0
-  fi
-  echo "$v (via Npm)"
-  
-  packages=$(find . -not -path "*/node_modules/*" -type f -name "package.json")
-  echo "$packages" | tr ' ' '\n' | while read -r item; do
-    echo "=========================="
-    echo "checking $item"
- 
-    case "$item" in
-    *"./package.json"* | *".github"* | *"functions/generated/client"*)
-      echo "ignoring $item"
-      continue
-      ;;
-    esac
-
-    cd "$(dirname "$item")/"
-
-    vCLI="$(node -e "$pkg;console.log(pkg.devDependencies['prisma'] || '')")"
-
-    if [ "$vCLI" != "" ]; then
-      if [ "$v" != "$vCLI" ]; then
-        echo "$item: prisma expected $v, actual $vCLI"
-
-        run_sync "$dir" "$branch"
-      fi
-
-      vPrismaClient="$(node -e "$pkg;console.log(pkg.dependencies['@prisma/client'])")"
-
-      if [ "$v" != "$vPrismaClient" ]; then
-        echo "$item: @prisma/client expected $v, actual $vPrismaClient"
-
-        run_sync "$dir" "$branch"
-      fi
-    else
-      echo "Dependency not found"
-    fi
-
-    cd "$dir"
-  done
-
-  echo ""
-  echo "=========================="
-  echo "after upgrade:"
-  git status
-
-  if [ -z "$(git status -s)" ]; then
-    echo "no changes"
-    end=$(date "+%s")
-    diff=$(echo "$end - $start" | bc)
-    remaining=$((interval - 1 - diff))
-    echo "took $diff seconds, sleeping for $remaining seconds"
-    sleep "$(no_negatives $remaining)"
-
-    continue
-  fi
```

Comment:
This logic here is pretty fancy, as it ensured that we did sleep at least long enough to achieve the wait time intended no matter how long the individual loop ran. No idea why we did it this way, or if this is important.

Reply:
I don't think it matters anymore since we loop through the projects only if there is an update, we also don't diff anymore as a result of that since we know there was an update.

Reply:
(can this be marked as resolved?)